### PR TITLE
77125f Update relationship origin page

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR updates the wording on the applicant relationship origin page to use customized wording and tense based on form data. Additionally, the applicant relationship page has been refactored to allow it to work for this second use case.

- Refactored applicant relationship page component
- Added relationship origin custom page
- Fixed a unit test
- Fixed usage of the `applicantWording` helper across the form
- Flipper: NA
- Team: IVC-CHAMPVA

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/77125

## Testing done

- Manual testing
- Verified existing tests run and pass

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |Screenshot 2024-03-13 at 19 07 10|Screenshot 2024-03-13 at 19 05 01|

## What areas of the site does it impact?

This form only

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

File: src/applications/ivc-champva/10-10D/pages/ApplicantMedicareStatusContinuedPage.jsx [hunk 1]
```diff
@@ -11,7 +11,9 @@ const keyname = 'applicantMedicarePart';
 function generateOptions({ data, pagePerItemIndex }) {
   const applicant = applicantWording(
     data?.applicants?.[pagePerItemIndex],
-  ).slice(0, -3); // remove 's_
+    undefined,
+    false,
+  );
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantMedicareStatusContinuedPage.jsx:16]: Updating the usage of this recently refactored `applicantWording` function. Not directly related to the issue at hand, but was causing strings to be truncated on other pages so I fixed it here and in two other places.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelOriginPage.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import ApplicantRelationshipPage, {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelOriginPage.jsx:1]: Custom page added - basically just a wrapper around `ApplicantRelationshipPage` but with a custom `generateOptions` function to give us the necessary custom wording for this screen.

**Review by cloudmagic80 — DISMISSED** [review]
LGTM

**Review by reiting — DISMISSED** [review]

**Review by balexandr — DISMISSED** [review]
Lgtm

**Comment by michaelclement** [comment]
Converting to draft because there are impending merge conflicts I want to get ahead of.

**Comment by jamigibbs** [comment]
Hi @michaelclemen, if you will still need a Platform FE CoP review for this after it's taken out of draft, let me know and I'll be happy to review.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx [hunk 3]
```diff
@@ -78,15 +103,16 @@ const relationshipStructure = {
 };
 
 export function ApplicantRelationshipReviewPage(props) {
-  const { data } = props || {};
+  const { data, keyname = KEYNAME } = props || {};
+  const genOps = props.genOp || generateOptions;
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:107]: This lets us target a different property on the applicant (i.e., relationship to sponsor vs relationship origin) and use a custom `generateOptions` function to produce the right custom radio button text while falling back to the original property and generate function if needed.

**Comment by michaelclement** [comment]
@jamigibbs conflicts have been ironed out and it's ready to be reviewed now. Thank you!

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**